testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -5594,6 +5619,90 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("array_union(cast(null as integer array), cast(null as integer array))");
   }
 
+  /** Tests {@code ARRAYS_OVERLAP} function from Spark. */
+  @Test void testArraysOverlapFunc() {
+    final SqlOperatorFixture f0 = fixture();
+    f0.setFor(SqlLibraryOperators.ARRAYS_OVERLAP);
+    f0.checkFails("^arrays_overlap(array[1, 2], array[2])^",
+        "No match found for function signature ARRAYS_OVERLAP\\("
+            + "<INTEGER ARRAY>, <INTEGER ARRAY>\\)", false);
+
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.SPARK);
+    f.checkScalar("arrays_overlap(array[1, 2], array[2])", true,
+        "BOOLEAN NOT NULL");
+    f.checkScalar("arrays_overlap(array[1, 2], array[3])", false,
+        "BOOLEAN NOT NULL");
+    f.checkScalar("arrays_overlap(array[1, null], array[1])", true,
```

Comment:
I want to confirm this `arrays_overlap(array[1, null], array[1])` the datatype is Boolean not BOOLEAN NOT NULL?

Reply:
yeap @NobiGo 
1) 
array[1, null] -> type is INT ARRAY NOT NULL
array[1] -> type is INT NOT NULL ARRAY NOT NULL
so the type is boolean

```
case class ArraysOverlap(left: Expression, right: Expression)
  override def nullable: Boolean = {
    left.nullable || right.nullable || leftArrayElementNullable || rightArrayElementNullable
  }
```

2) spark/postgres's typeof doesn't display not null/null info, but calcite/flink does. so the spark display it not full.
we should confirm the behavior by code

 you can look here https://issues.apache.org/jira/browse/FLINK-31906

```
spark-sql (default)> select typeof(arrays_overlap(array(1, null), array(1)));
boolean

spark-sql (default)> select typeof(arrays_overlap(array(1), array(1)));
boolean // don't display not null info.
```

Reply:
@liuyongvs Maybe we can test in Spark By:
`arrays_overlap(array[1, null], array[1]) IS NOT NULL` ?

Reply:
@NobiGo can not.
IS NOT NULL just judge whether the value is null
```
   for example arrays_overlap(array[1, null], array[1])  is true, so it is not null. then the result arrays_overlap(array[1, null], array[1]) IS NOT NULL is true
```
while the nullable is a constraint, just for whether it can store not null/null value.
the semantic is different